src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -8,25 +8,22 @@
 import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
 import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
 
-import java.util.Collections;
-import java.util.HashSet;
 import java.util.List;
 import java.util.Objects;
 import java.util.Optional;
-import java.util.Set;
 
 import seedu.address.commons.core.index.Index;
 import seedu.address.commons.util.CollectionUtil;
 import seedu.address.commons.util.ToStringBuilder;
 import seedu.address.logic.Messages;
 import seedu.address.logic.commands.exceptions.CommandException;
 import seedu.address.model.Model;
-import seedu.address.model.person.Address;
-import seedu.address.model.person.Email;
-import seedu.address.model.person.Name;
 import seedu.address.model.person.Person;
-import seedu.address.model.person.Phone;
-import seedu.address.model.tag.Tag;
+import seedu.address.model.person.fields.mandatory.Address;
+import seedu.address.model.person.fields.mandatory.Email;
+import seedu.address.model.person.fields.mandatory.Name;
+import seedu.address.model.person.fields.mandatory.Phone;
+import seedu.address.model.person.fields.optional.Tags;
```

Comment:
Should the fields be put under "mandatory"/"optional" subpackages? Shoudn't packages should be split by subjects instead of adverbs?

Reply:
Agreed, I will remove them in the subsequent commit. For optional fields, `Person` will have an `Optional<Field>` instead of a `Field`. For optional fields which are based on a `Collection` (such as `Tags`), `Person` will have `Field` in which the `Collection` may be empty. This will help to minimize problems with `null`.